### PR TITLE
armpl-gcc: add v23.10 and macOS support

### PR DESCRIPTION
**Pull request by annop-w**

### CODE REVIEW

File: var/spack/repos/builtin/packages/armpl-gcc/package.py [hunk 1]
```diff
@@ -177,28 +228,36 @@
...
+    compiler_version = version.split("_", 1)[1]
     os = get_os(armpl_version)
-    filename = "arm-performance-libraries_" + armpl_version + "_" + os + "_" + gcc_version + ".tar"
+    if os == "macOS":
+        if armpl_version.startswith("23.06"):
+            return f"{base_url}{armpl_version_dashed}/armpl_{armpl_version}_{compiler_version}.dmg"
+        else:
+            filename = f"arm-performance-libraries_{armpl_version}_macOS.dmg"
+            return f"{base_url}{armpl_version_dashed}/macos/{filename}"
+    filename = "arm-performance-libraries_{}_{}_{}.tar".format(armpl_version, os, compiler_version)
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/armpl-gcc/package.py]: Thanks for switching to f-strings.  Curious why this one still uses `format()`.  Perhaps I'm missing something?

**Review by tldahlgren — DISMISSED** [review]
Confirmed the version sha256 for RHEL8, rest LGTM.  Just a minor, non-blocking question.

**Comment by tldahlgren** [comment]
@annop-w ping me when you're ready for this PR to be merged.

File: var/spack/repos/builtin/packages/armpl-gcc/package.py (hunk 1 above)
Comment (annop-w) [reply var/spack/repos/builtin/packages/armpl-gcc/package.py]: @tldahlgren There are a number of places that still use `format()` in this package. I can replace all those in another patch.

**Comment by annop-w** [comment]
@tldahlgren I've replaced all `format()` with f-strings now. Could you please take a look again ? Thank you.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**